transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java

```diff
@@ -457,56 +458,69 @@ private void registerCallbacks(
         callbackHandlerMap.put(AuthenticationTokenCallback.class, authenticationTokenCallbackHandler);
 
         String forwarded = httpRequest.getHeader(HEADER_FORWARDED);
```

Comment:
Two cases:
1. client --> gateway
In this case, there is no Forwarded header, we just use session's remote address. In fact, the request wouldn't even go through http.proxy service.

* client --> gateway1 ---> gateway2
In this case, there is Forwarded header from gateway1. The filter at gateway2 should use Forwarded header. In this case, the request would go through http.proxy service of gateway1 which adds Forwarded header (if configured to inject)

Hence, I am suggesting to use Forwarded header (if there is one), otherwise just use session's remote address(the best guess of client's address)